compiler-core/src/parse.rs

```diff
@@ -2207,8 +2209,9 @@ where
                         as_name: None,
                     };
                     if self.maybe_one(&Token::As).is_some() {
-                        let (_, as_name, _) = self.expect_upname()?;
+                        let (_, as_name, end) = self.expect_upname()?;
                         import.as_name = Some(as_name);
+                        import.location.end = end;
```

Comment:
Won't this change how the errors for duplicate imports etc are shown? I would like to keep them as only the first part of the import and not `.{}` etc

Reply:
I made this change so that the `as name` would be included as part of the import so that it would also have the goto work properly. this would change the highlight to include the `as name` part of the import but not the `.{}`. That being said I'm fine with flipping this back and just not having the goto work on the rename part of the import

Reply:
Sorry, I'm not following. `as name` comes after `.{}` so wouldn't it be included?

Let's add a test for an invalid import or something where the import is over multiple lines, that'll make it clear what is happening.

Reply:
This change was for unqualified `as name` for an individual type/value like `import example_module.{my_num as num}`. Lemme double check if we have a test for errors with the wider `as name` and the unqualified `as name` and add them if they are missing

Reply:
Ok I added the tests though looking at the output more, I wonder if making this change would be better than the existing behavior for unqualified imports? Cause the actual issue is technically not that the type is imported twice but that the same name is used with the as. Lmk what you think

<img width="903" alt="Screenshot 2024-04-30 at 11 12 17 AM" src="https://github.com/gleam-lang/gleam/assets/5996838/e9ffd9c4-4661-49cf-94b9-239a275f4c5f">
<img width="401" alt="Screenshot 2024-04-30 at 11 12 26 AM" src="https://github.com/gleam-lang/gleam/assets/5996838/98a31b22-d21d-4bf0-9221-614200bab877">

Reply:
Ah yes! Sorry I got confused there.

I agree, the error could be improved here